include/bluetooth/bluetooth.h

```diff
@@ -398,6 +471,130 @@ int bt_le_adv_update_data(const struct bt_data *ad, size_t ad_len,
+ *			  is already maximum number of connections established.
+ *			  This error code is only guaranteed when using Zephyr
+ *			  controller, for other controllers code returned in
+ *			  this case may be -EIO.
+ */
+struct bt_adv *bt_le_adv_set_create(const struct bt_le_adv_param *param);
+
+/** Start advertising with the given advertising set
+ *
+ *  Start advertising with a specific advertising set.
+ *  If both timeout and advertising event limit is given then the advertising
+ *  set will be stopped once one of them is reached.
+ *
+ *  @param adv         Advertising set object.
+ *  @param timeout     Timeout in ms, K_FOREVER or K_NO_WAIT will disable timeout.
+ *  @param num_events  Stop advertising after the specified number of advertising
+ *                     events has been reached.
+ */
+int bt_le_adv_set_start(struct bt_adv *adv, s32_t timeout,
+			u8_t num_events);
+
+/** Stop advertising with the given advertising set
+ *
+ *  Stop advertising with a specific advertising set.
+ *
+ *  @param adv Advertising set object.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_adv_set_stop(struct bt_adv *adv);
+
+/** Set advertising set advertising or scan response data.
+ *
+ *  Set advertisement data or scan response data. If the advertising set is
+ *  currently advertising then the advertising data will be updated in
+ *  subsequent advertising events.
+ *
+ *  If the advertising set has been configured with Primary advertising then the
+ *  maximum data length is 31 bytes. If the advertising set has been configured
+ *  for extended advertising data length, then the maximum data length defined
+ *  by the controller.
+ *
+ *  NOTE: Not all scanners support extended data length advertising data.
+ *
+ *  @param ad      Data to be used in advertisement packets.
+ *  @param ad_len  Number of elements in ad
+ *  @param sd      Data to be used in scan response packets.
+ *  @param sd_len  Number of elements in sd
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_adv_set_data(struct bt_adv *adv,
+		       const struct bt_data *ad, size_t ad_len,
+		       const struct bt_data *sd, size_t sd_len);
+
+/** Delete advertising set.
+ *
+ *  Delete advertising set. This will free up the advertising set and make it
+ *  possible to create a new advertising set.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_le_adv_set_delete(struct bt_adv *adv);
+
+/** @brief Get array index of a advertising set.
+ *
+ *  This function is used to map bt_adv to index of an array of
+ *  advertising sets. The array has CONFIG_BT_MAX_ADV elements.
+ *
+ *  @param adv Advertising set.
+ *
+ *  @return Index of the advertising set object.
+ *  The range of the returned value is 0..CONFIG_BT_MAX_ADV-1
+ */
+u8_t bt_adv_index(struct bt_adv *adv);
+
+/** @brief Advertising set info structure. */
+struct bt_adv_info {
+
+	/* ID Address used for advertising */
```

Comment:
Can you add a bit of info here on what function to use to retrieve the address from the ordinal?

Reply:
"Which local identity the advertising set is using" Is that a better explanation?
The address is below. Maybe it should just be addr. And put the public identity address there if the advertiser is using it.